org.eclipse.sprotty/src/main/java/org/eclipse/sprotty/DefaultDiagramServer.java

```diff
@@ -193,6 +193,11 @@ public void dispatch(Action action) {
 		Consumer<ActionMessage> remoteEndpoint = getRemoteEndpoint();
 		if (remoteEndpoint != null) {
 			remoteEndpoint.accept(new ActionMessage(getClientId(), action));
+			if (action instanceof SelectAction) {
+				handle((SelectAction) action);
+			} else if (action instanceof SelectAllAction) {
+				handle((SelectAllAction) action);
```

Comment:
The `handle` methods are meant for incoming actions. Forwarding outgoing actions to them could cause confusion, and we would have to do that for other actions, too, for completeness (e.g. CollapseExpandAction). I would prefer not to add this special handling.

Reply:
I agree that using `handle` is probably not the best solution to the problem. If I extracted the useful logic into another method that is used in both `handle` and here, would that be okay?

Reply:
ok

Reply:
I made some changes so now I call `updateSelection` in both `handle` and `dispatch`. Since you were talking about the `CollapseExpandAction`, should I make similar changes for those actions as well?

Reply:
I think it would be better to have a working example first.

Reply:
okay